clippy_lints/src/methods/mod.rs

```diff
@@ -2150,18 +2150,19 @@ fn lint_clone_on_ref_ptr(cx: &LateContext<'_>, expr: &hir::Expr<'_>, arg: &hir::
             return;
         };
 
+        let snippet = if in_macro(arg.span) {
+            snippet_with_macro_callsite(cx, arg.span, "_")
+        } else {
+            snippet(cx, arg.span, "_")
+        };
```

Comment:
```suggestion
        let snippet = snippet_with_macro_callsite(cx, arg.span, "_");
```
This can be simplified because `Span::source_callsite()` takes care of doing that check internally.

Reply:
Neat ! ❤️